darts/timeseries.py

```diff
@@ -3,117 +3,554 @@
 
-        return columns
+        Returns
+        -------
+        TimeSeries
+            A univariate and deterministic TimeSeries constructed from the inputs.
+        """
 
-    def _assert_univariate(self):
+        df = pd.DataFrame(pd_series)
+        return TimeSeries.from_dataframe(df,
+                                         time_col=None,
+                                         value_cols=None,
+                                         fill_missing_dates=fill_missing_dates,
+                                         freq=freq)
+
+    @staticmethod
+    def from_times_and_values(times: Optional[Union[pd.DatetimeIndex, pd.RangeIndex]],
+                              values: Union[np.ndarray, pd.DataFrame],
+                              fill_missing_dates: Optional[bool] = True,
+                              freq: Optional[str] = None,
+                              columns: Optional[pd._typing.Axes] = None) -> 'TimeSeries':
+        """
+        Returns a deterministic TimeSeries built from an index and values.
+
+        Parameters
+        ----------
+        times
+            A `pandas.DateTimeIndex` or `pandas.RangeIndex` representing the time axis for the time series.
+        values
+            A 2-dimensional Numpy array of values for the TimeSeries. The dimensions should be (time, component).
+        fill_missing_dates
+            Optionally, a boolean value indicating whether to fill missing dates with NaN values. This requires
+            either a provided `freq` or the possibility to infer the frequency from the provided timestamps.
+        freq
+            Optionally, a string representing the frequency of the Pandas DataFrame. This is useful in order to fill
+            in missing values if some dates are missing and `fill_missing_dates` is set to `True`.
+        columns
+            Columns to be used by the underlying pandas DataFrame.
+
+        Returns
+        -------
+        TimeSeries
+            A TimeSeries constructed from the inputs.
+        """
+
+        # TODO: make it more general, not always going via a DataFrame
+
+        idx = times if isinstance(times, pd.RangeIndex) or isinstance(times, pd.DatetimeIndex) else None
+        df = pd.DataFrame(values, index=idx)
+        if columns is not None:
+            df.columns = columns
+
+        return TimeSeries.from_dataframe(df,
+                                         time_col=None,
+                                         value_cols=None,
+                                         fill_missing_dates=fill_missing_dates,
+                                         freq=freq)
+
+    @staticmethod
+    def from_values(values: np.ndarray,
+                    fill_missing_dates: Optional[bool] = True,
+                    freq: Optional[str] = None,
+                    columns: Optional[pd._typing.Axes] = None) -> 'TimeSeries':
+
+        time_index = pd.RangeIndex(0, len(values), 1)
+
+        if len(values.shape) <= 2:
```

Comment:
I made "times" mandatory in "from_times_and_values()", so I think we could leave it like that, WDYT?
This method should work and it will leverage "from_times_and_values()" when dimensionality is 2.